src/AutorestSwift/View Models/KeyValueViewModel.swift

```diff
@@ -157,12 +165,14 @@ struct KeyValueViewModel {
         Create a ViewModel with a Key and Value pair
         - Parameter key: Key String in the Key value pair
         - Parameter value: the value string
+        - Parameter optional: a flag indicates if the Key/Value pair is optional
+        - Parameter implementation: an enum indicates if the value is from Client property or from Options struct
      */
-    init(key: String, value: String) {
+    init(key: String, value: String, optional: Bool = false, path: String? = nil) {
         self.key = key
         self.value = value
-        self.optional = false
-
+        self.optional = optional
+        self.path = path ?? "options.\(key)"
```

Comment:
The special case here makes me uneasy. If I have a key value, I feel that the path should point to the value, not the key. This might work since usually the key property and the  value property have the same name.

Reply:
Removed.